### ISSUE TITLE
 vcpkg was unable to detect the active compiler's information. See above for the CMake failure output. Failure in Manifest mode. 

### ISSUE THREAD

**Issue by Suresh3d** [microsoft/vcpkg#27662]
**Host Environment**

- OS: windows 10, 64 bit
- Compiler: visual studio 2019

**To Reproduce**

1) I cloned vcpkg . ran bootstrap bat file . then did vcpkg integrate install. 
2) I created vcpkg.json in my project folder. Here is my vcpkg.json file content. 

```
{
    "name": "versions-test",
    "version": "1.0.0",
    "dependencies": [
        {
            "name": "curl",
            "version>=": "7.74.0#5"
        } 
    ],
	"builtin-baseline": "a4013afe6d3053e739b37c4cb6b5e64b55611e24"
}
```
3) I build my project . It says suitable version of cmake is not found , so then downloading . at the end it throws error ,  "vcpkg was unable to detect the active compiler's information. See above for the CMake failure output"
 

**Failure logs**

```
1>------ Build started: Project: openJob, Configuration: Release x64 ------
1>Installing vcpkg dependencies to D:\AMN\amn-nx-plugin\vcpkg_installed\x64-windows\
1>"D:\vcpkg\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows" --vcpkg-root "D:\vcpkg\\" "--x-manifest-root=D:\AMN\amn-nx-plugin\\" "--x-install-root=D:\AMN\amn-nx-plugin\vcpkg_installed\x64-windows\\"
1>A suitable version of cmake was not found (required v3.24.0). Downloading portable cmake v3.24.0...
1>Downloading cmake...
1>  https://github.com/Kitware/CMake/releases/download/v3.24.0/cmake-3.24.0-windows-i386.zip -> D:\vcpkg\downloads\cmake-3.24.0-windows-i386.zip
1>Extracting cmake...
1>Detecting compiler hash for triplet x64-windows...
1>A suitable version of powershell-core was not found (required v7.2.6). Downloading portable powershell-core v7.2.6...
1>Downloading powershell-core...
1>  https://github.com/PowerShell/PowerShell/releases/download/v7.2.6/PowerShell-7.2.6-win-x86.zip -> D:\vcpkg\downloads\PowerShell-7.2.6-win-x86.zip
1>Extracting powershell-core...
1>D:\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets(158,5): warning : error: while detecting compiler information:
1>The log file content at "D:\vcpkg\buildtrees\detect_compiler\stdout-x64-windows.log" is:
1>-- Downloading https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-win.zip -> ninja-win-1.10.2.zip...
1>-- Configuring x64-windows
1>CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
1>    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
1>    Working Directory: D:/vcpkg/buildtrees/detect_compiler/x64-windows-rel/vcpkg-parallel-configure
1>    Error code: 1
1>    See logs for more information:
1>      D:\vcpkg\buildtrees\detect_compiler\config-x64-windows-CMakeCache.txt.log
1>      D:\vcpkg\buildtrees\detect_compiler\config-x64-windows-out.log
1>
1>Call Stack (most recent call first):
1>  scripts/cmake/vcpkg_configure_cmake.cmake:311 (vcpkg_execute_required_process)
1>  scripts/detect_compiler/portfile.cmake:18 (vcpkg_configure_cmake)
1>  scripts/ports.cmake:147 (include)
1>
1>
1>
1>D:\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets(158,5): warning : error: vcpkg was unable to detect the active compiler's information. See above for the CMake failure output.
1>D:\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets(158,5): error MSB3073: The command ""D:\vcpkg\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows" --vcpkg-root "D:\vcpkg\\" "--x-manifest-root=D:\AMN\amn-nx-plugin\\" "--x-install-root=D:\AMN\amn-nx-plugin\vcpkg_installed\x64-windows\\" " exited with code 1.
1>Done building project "openJob.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Additional context**

I am not using cmake project in visual studio . Mine is just normal C++ project. I don't know why vcpkg is downloading cmake for this.
 
Previously I cloned vcpkg to sub folders , but then I cloned directly to D drive . It doesn't help. 

[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/9938759/config-x64-windows-out.log)
[config-x64-windows-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/9938760/config-x64-windows-CMakeCache.txt.log)

**Comment by Neumann-A** [comment]
> Compiler: visual studio 2019
`"-DVCPKG_PLATFORM_TOOLSET=v141"`

something is not fitting together here `VS 2019` should be `v142`. vcpkg isn't passing a toolset by default so did you set `VCPKG_PLATFORM_TOOLSET` in the triplet?

**Comment by Suresh3d** [comment]
@Neumann-A No , I didn't set anything. Actually I noticed that normally installing libraries from cmd is also not working. I already installed vcpkg a year ago in my machine . I used to install libraries from cmd using "vcpkg install packagname" . Now I wanted to port to my project to other machine for other developer . in his machine vcpkg latest version is installing latest version of packages . 
So I tried to clone vcpkg now in different location. I didn't update last one because if I don't want to break existing one if anything happens. In VS project folder I created vcpkg.json file and tried to build. It throws this error . Then I checked the latest installation directory , normal installation through cmd is also throwing same error.  
Note : I cloned latest vcpkg in my colleague machine , and installing packages from cmd is working fine there . I don't know what was changed in my project.
